### ISSUE TITLE
Refactoring: Simplify the logic

### ISSUE THREAD

**Issue by furqanbaqai** [furqanbaqai/RockPaperScissorsContract#7]
I suspect you can also simplify the conditional logic a lot. Consider a game goes a few phases. A new game is created by someone who could be player1. Player1's bet and move could be part of the new game flow. Then, the main thing that should happen next, is player2 commits, which finishes the game. Next, player 1 either reveals or forfeits. You could make it so both players need to send the right amount of ETH, and player1 can recall the bet in the exception where player2 doesn't show up.

In other words, you could make the functions more explicit about who should use them, like, `player1CreateGameAndBet`, `player2JoinGameAndBet`, `player1RevealAndFinalizeGame`. **Notice, player2 doesn't need to reveal? Why hide the move when player1 is committed? Also, player1CancelGame to get money back if there is no one to play against. And ... the usual functions to deposit and withdraw ETH to/from the contract which you know very well.**

**Comment by furqanbaqai** [comment]
Done.
